### PR TITLE
Hive: TestHiveIcebergStorageHandler should not run all of the tests in every combination (#1924)

### PR DESCRIPTION
**Pull request by pvary**
As mentioned in #1924 this PR separates out the test cases needing a different combination of parameters

### CODE REVIEW

File: mr/src/test/java/org/apache/iceberg/mr/hive/TestHiveIcebergStorageHandlerLocalScan.java [hunk 1]
```diff
@@ -0,0 +1,553 @@
...
+public class TestHiveIcebergStorageHandlerLocalScan {
+
+  @Parameters(name = "fileFormat={0}, catalog={1}")
+  public static Collection<Object[]> parameters() {
+    Collection<Object[]> testParams = new ArrayList<>();
+    for (FileFormat fileFormat : HiveIcebergStorageHandlerTestUtils.FILE_FORMATS) {
+      for (TestTables.TestTableType testTableType : TestTables.ALL_TABLE_TYPES) {
+        testParams.add(new Object[] {fileFormat, testTableType});
+      }
+    }
```
Comment (rdblue) [inline mr/src/test/java/org/apache/iceberg/mr/hive/TestHiveIcebergStorageHandlerLocalScan.java:71]: Do we need tests for each type with each format? Seems like running tests across all formats and then across all types would work fine. All of the tables use `BaseTable`, so the table types are really just testing catalogs. Could we separate out the catalog tests?

File: mr/src/test/java/org/apache/iceberg/mr/hive/TestHiveIcebergStorageHandlerWithEngine.java [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+                  Types.FloatType.get(), Types.DoubleType.get(), Types.DateType.get(), Types.TimestampType.withZone(),
+                  Types.TimestampType.withoutZone(), Types.StringType.get(), Types.BinaryType.get(),
+                  Types.DecimalType.of(3, 1));
+
+  @Parameters(name = "fileFormat={0}, engine={1}, catalog={2}")
+  public static Collection<Object[]> parameters() {
+    String javaVersion = System.getProperty("java.specification.version");
+
+    Collection<Object[]> testParams = new ArrayList<>();
+    for (FileFormat fileFormat : HiveIcebergStorageHandlerTestUtils.FILE_FORMATS) {
```
Comment (rdblue) [inline mr/src/test/java/org/apache/iceberg/mr/hive/TestHiveIcebergStorageHandlerWithEngine.java:76]: Here as well, one format should be sufficient. We want to know that all formats work through Hive, not that all formats work when loaded by Hive through a HadoopCatalog table.

**Comment by rdblue** [comment]
Looks like this is really effective at reducing build time! I'd still like to eliminate a lot of the tests that aren't adding much value. We should have tests for Hive reading formats and tests for Hive working with tables, but we don't need all the format tests run on all the tables, or all the table tests run across all formats and engines.

File: mr/src/test/java/org/apache/iceberg/mr/hive/TestHiveIcebergStorageHandlerLocalScan.java (hunk 1 above)
Comment (pvary) [reply mr/src/test/java/org/apache/iceberg/mr/hive/TestHiveIcebergStorageHandlerLocalScan.java:78]: Changed so we run: HiveCatalog with all of the FileFormats, and then we run every other Catalog with Parquet

File: mr/src/test/java/org/apache/iceberg/mr/hive/TestHiveIcebergStorageHandlerWithEngine.java (hunk 2 above)
Comment (pvary) [reply mr/src/test/java/org/apache/iceberg/mr/hive/TestHiveIcebergStorageHandlerWithEngine.java:77]: Change so we run HiveCatalog with every FileFormat and every Engine, then run every other catalog with Parquet and mr

**Comment by rdblue** [comment]
JDK 8 tests run in 36 minutes! Great work, @pvary!

**Comment by pvary** [comment]
Thanks for the review and the merge @rdblue!

With this runtimes do you think we still need to separate out the integration tests?
We might just keep everything as it is for now for simplicity shake, and if we see higher runtimes later then we can revisit the integration test idea?

**Comment by rdblue** [comment]
> With this runtimes do you think we still need to separate out the integration tests?

No, I think this is okay now. If we wanted to run all combinations, I'd say that is good for integration tests. But as long as we're careful about what we're testing and reducing the combinations if they don't provide much value we are good.

**Pull request merged**